Create program page

myang1010 opened the pull request:
What (if any) features are you implementing?
- Setup the program detail page

What (if anything) did you refactor?
- Removed a comma in Review page

Is there anything that you need from your teammate?
- Not sure where does the Get Help, Apply Now should redirect to
- Also the `Save this result` button doesn't have any function in it

Any other comments, questions, or concerns?
- `Estimated Annual Value` and `Estimated Time to Apply` text, and  `Get Help`, `Apply Now` and `SAVE THIS RESULT` button names currently don't have any translation.

Screenshots

- desktop, large screen
<img width="734" alt="Screenshot 2024-01-31 at 2 03 08 PM" src="https://github.com/Gary-Community-Ventures/benefits-calculator/assets/29790405/c6e600d4-738f-411d-bee4-1ca48c06f590">



- mobile, small screen
<img width="272" alt="Screenshot 2024-01-31 at 1 55 39 PM" src="https://github.com/Gary-Community-Ventures/benefits-calculator/assets/29790405/f93c325a-2764-4c7c-8e22-112d197ff88b">

sydney-devine reviewed (changes requested):
I'm the worst. Realizing that the link in the ticket was the original wireframe link and we had updated it slightly. This is totally my bad, and I'm so sorry I didn't catch that. 

https://xd.adobe.com/view/9c25b118-90b5-409e-b9b6-36662f2b59c2-c976/screen/2341183d-f7d8-4568-8ec9-32c96d71b5c3/?fullscreen

Changes visually from what I'm seeing:
- Where it says Medicaid in the orange at the top, that should be pulling from the category heading. Want to make sure it's doing that. 
- Remove the "Get Help" button under the estimated value/time to apply box
- Add in Get Help Applying box, light lavender background. The information that pulls into this box should come from the navigators info on the backend. @CalebPena can you help specify where that might be found? 
- Save my results button should go back on top right. But I think that might be addressed with the work that @ivonne-hernandez is doing for those top navigational buttons. 
- The current "Back to Results" button, can we left alight it with the estimated annual value / time to apply box so it isn't floating so far to the left?

CalebPena commented:
Navigators is the `program.navigators` field.

myang1010 commented:
@sydney-devine and @CalebPena thanks for the feedback. A few more questions regarding the updated design doc, 

<img width="355" alt="Screenshot 2024-01-24 at 3 39 08 PM" src="https://github.com/Gary-Community-Ventures/benefits-calculator/assets/29790405/77e286be-f210-48d1-8bb7-a2df4b61aa25">

1. The estimated time we have by default is like `45 minutes` while the design doc is `45 Min`, do we want exact match with the design doc?
2. Do we have the question mark icon that I can use?
3. For the `program.navigators` seems they'll come in an array. Do we know the data format of each element inside the navigators array? Looks like there there will be some sort of organization name, address and contact info. Is it possible that I can get some mock data for this field(Not a blocker, but nice to have)?

Thanks.

sydney-devine commented:
> @sydney-devine and @CalebPena thanks for the feedback. A few more questions regarding the updated design doc,
> 
> <img alt="Screenshot 2024-01-24 at 3 39 08 PM" width="355" src="https://private-user-images.githubusercontent.com/29790405/299493216-77e286be-f210-48d1-8bb7-a2df4b61aa25.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDYyMDEwODksIm5iZiI6MTcwNjIwMDc4OSwicGF0aCI6Ii8yOTc5MDQwNS8yOTk0OTMyMTYtNzdlMjg2YmUtZjIxMC00OGQxLThiYjctYTJkZjRiNjFhYTI1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTI1VDE2Mzk0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkxYTA0MjY2NThkZTYwYTJmNjUwY2FjNmVkODY5MTc3NDRhOTRkODhjMDJlY2Q1NzQxMDVmZDkzNmZiOTQ5YmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.PiCGiVJULZ_H1DPmfpdiiemW187GNoLcvgukfHtbnPs">
> 1. The estimated time we have by default is like `45 minutes` while the design doc is `45 Min`, do we want exact match with the design doc?
> 2. Do we have the question mark icon that I can use?
> 3. For the `program.navigators` seems they'll come in an array. Do we know the data format of each element inside the navigators array? Looks like there there will be some sort of organization name, address and contact info. Is it possible that I can get some mock data for this field(Not a blocker, but nice to have)?
> 
> Thanks.

Hi @myang1010 

1. I would say let's go with "min". I believe that's something we can control on the backend by updating those values to just say 45 min though. @CalebPena can you confirm that's something we can just update? 
2. Ignore the question mark icon for now. It's going to go with the help bubble ticket. That's just mocked up there as a potential for what it could look like. 
3. @CalebPena are you able to provide this for Ming?

CalebPena commented:
@myang1010 The estimated time is a string sent from the backend, so we can just update it in the backend. Also, you can get some mock navigator data if you go to Django admin, go to the navigators tab and click on a navigator, and then select some programs for that navigator.

![image](https://github.com/Gary-Community-Ventures/benefits-calculator/assets/62856626/be9bd785-43bf-4158-8e58-b4ef12ddf78b)

CalebPena commented:
FYI, the Results top buttons pr is merged, so you can use it in this pr.

ivonne-hernandez commented on src/Components/Results/ProgramPage/ProgramPage.tsx:
```diff
@@ -1,24 +1,104 @@
+            <li key={index} className="apply-info">
+              <ResultsTranslate translation={info.name} />
+              <address className="address-info">
+                <p>Address bla bla bla</p>
```
What do you think about adding the `info.address` here instead of `Address bla bla bla`?

ivonne-hernandez commented on src/Components/Results/ProgramPage/ProgramPage.tsx:
```diff
@@ -1,24 +1,104 @@
+              <ResultsTranslate translation={info.name} />
+              <address className="address-info">
+                <p>Address bla bla bla</p>
+                <a href={info.assistance_link.default_message}>Visit Website</a>
```
Can you please wrap the `Visit Website` text in a `FormattedMessage`?

ivonne-hernandez commented on src/Components/Results/ProgramPage/ProgramPage.tsx:
```diff
@@ -1,24 +1,104 @@
+                <br />
+                <a href={`mailto:${info.email}`}>{info.email.default_message}someone@somewhere.com</a>
+                <br />
+                <a href={`tel:${info.phone_number}`}>{info.phone_number}123.123.1233</a>
```
Same here. What do you think about removing `123.123.1233`?

ivonne-hernandez commented on src/Components/Results/ProgramPage/ProgramPage.tsx:
```diff
@@ -1,24 +1,104 @@
+      </section>
+
+      <section className="required-docs">
+        <h3>Required Documents Checklist</h3>
```
Can you please wrap this text in a `FormattedMessage`?

ivonne-hernandez reviewed (changes requested):
Please see comments above for requested changes.

myang1010 commented:
Thank you @ivonne-hernandez and @CalebPena for the feedback.

1. I've removed those unnecessary text for email and phone number. 

   I'm keeping the address info as I don't see the `address` field from `navigator`. Please let me know if I'm using the wrong input. Here's what I have, 

    ><img width="501" height="550" alt="Screenshot 2024-02-08 at 2 18 40 PM" src="https://github.com/Gary-Community-Ventures/benefits-calculator/assets/29790405/3ab1fb8b-1781-4142-86e6-28c1e9c6add2">

2. I've wrapped "Visit Website" text and "Required Documents Checklist" text in FormattedMessage. But I'm not sure about the correct ids for those two. I'm giving them `id="results.visit-webiste"` and `id="results.required-documents-checklist"` for now. Also unsure about id for "BACK TO RESULTS" button, so giving it `id="results.back-to-results-btn"`.
3. I also noticed the "Warning(69:3) autoprefixer: start value has mixed support, consider using flex-start instead". It's from `.MuiFormControlLabel-root.vertical-align` class in Filter.css. If it hasn't been taken cared of, I can change it to `flex-start` in this PR.

CalebPena commented on src/Components/Results/ProgramPage/ProgramPage.tsx:
```diff
@@ -1,24 +1,115 @@
+                  <FormattedMessage id="results.visit-webiste" defaultMessage="Visit Website" />
+                </a>
+                <br />
+                <a href={`mailto:${info.email}`}>{info.email.default_message}</a>
```
We support having a different email depending on the language, could you translate `info.email`

CalebPena commented on src/Components/Results/ProgramPage/ProgramPage.tsx:
```diff
@@ -1,24 +1,115 @@
+          {program.navigators.map((info, index) => (
+            <li key={index} className="apply-info">
+              <ResultsTranslate translation={info.name} />
+              <address className="address-info">
```
Why did you put all of the information in the `address` element? I am pretty sure that the `address` element is should only be used for actual addresses

CalebPena commented on src/Components/Results/ProgramPage/ProgramPage.tsx:
```diff
@@ -1,24 +1,115 @@
+              <ResultsTranslate translation={info.name} />
+              <address className="address-info">
+                <p>Address info goes here</p> {/* replace with info.address */}
+                <a href={info.assistance_link.default_message}>
```
`email`, `phone`, and `assistance_link` can be blank, if a navigator does not have one of those things. Could you make it so that if there is no default message, then the field does not show up.

CalebPena commented on src/Components/Results/ProgramPage/ProgramPage.tsx:
```diff
@@ -1,24 +1,115 @@
+        <h3>
+          <FormattedMessage id="results.required-documents-checklist" defaultMessage="Required Documents Checklist" />
+        </h3>
+        <ul className="required-docs-list">
+          {program.documents.map((document, index) => (
+            <li key={index}>
+              <ResultsTranslate translation={document} />
+            </li>
+          ))}
+        </ul>
```
Documents can be an empty array, could you not render this if there are no documents.

sydney-devine reviewed (changes requested):
Can we update the desktop version of it so that line between the category name and the benefit name doesn't extend so far past everything else?

CalebPena commented on src/Components/Results/ProgramPage/ProgramPage.tsx:
```diff
@@ -1,24 +1,115 @@
+
+      <div className="apply-online-button">
+        <Link to={program.apply_button_link.default_message} target="_blank" rel="noopener noreferrer">
+          Apply Online
```
All plane text should be in a `FormattedMessage`, so that it can be translated.

CalebPena commented on src/Components/Results/ProgramPage/ProgramPage.tsx:
```diff
@@ -1,24 +1,115 @@
+            <li key={index} className="apply-info">
+              <ResultsTranslate translation={info.name} />
+              <address className="address-info">
+                <p>Address info goes here</p> {/* replace with info.address */}
```
I think that we can get rid of this for now, because we don't currently have a backend field for addresses.

myang1010 commented:
@CalebPena, in this latest commit, I've converted all measurements from pixels (px) to rems (rem). I've also made adjustments so that navigators or documents are only rendered when there's a 'default_message' available. Additionally, I've wrapped all plain text within either FormattedMessage or ResultsTranslate components. Also website URLs, email addresses, and phone numbers can be placed within an `address` tag, although it's not strictly necessary. Replaced it with regular `div`. 

@sydney-devine , here's the updated screenshots,

<img width="1295" alt="no_nav_doc" src="https://github.com/Gary-Community-Ventures/benefits-calculator/assets/29790405/d8fd7427-f75b-4d87-a69a-d9832096abce">
<img width="871" alt="only_doc" src="https://github.com/Gary-Community-Ventures/benefits-calculator/assets/29790405/dd836f3c-e6f2-45c4-9e21-8cea2912af92">
<img width="275" alt="small_screen" src="https://github.com/Gary-Community-Ventures/benefits-calculator/assets/29790405/7835c6f1-9517-4b39-86f1-9f749b18b04b">

Thank you all for the feedback. Please let me know if there's any other concerns. Thanks!

sydney-devine reviewed (approved):
Visuals look good on my end
